File: AndroidApp/app/src/main/java/com/example/greengrowtechapp/Handlers/Pot.java
package com.example.greengrowtechapp.Handlers;

import com.google.gson.annotations.SerializedName;

public class Pot {

    @SerializedName("potId")
    private int potId;

    @SerializedName("potName")
    private String potName;

    @SerializedName("potType")
    private int potType;

    @SerializedName("plantName")
    private String plantName;

    @SerializedName("userId")
    private int userId;

    @SerializedName("hasCamera")
    private boolean hasCamera;

    @SerializedName("pictReq")
    private boolean pictReq;

    @SerializedName("pumpStatus")
    private boolean pumpStatus;

    @SerializedName("greenHouseStatus")
    private boolean greenHouseStatus;

    @SerializedName("greenHouseTemperature")
    private double greenHouseTemperature;

    @SerializedName("greenHouseHumidity")
    private double greenHouseHumidity;

    @SerializedName("greenHousePressure")
    private double greenHousePressure;

    @SerializedName("potPotassium")
    private double potPotassium;

    @SerializedName("potPhospor")
    private double potPhosphorus;

    @SerializedName("potNitrogen")
    private double potNitrogen;

    public Pot(int potId, String potName, int potType, String plantName, int userId, boolean hasCamera, boolean pictReq,
               boolean pumpStatus, boolean greenHouseStatus, double greenHouseTemperature, double greenHouseHumidity,
               double greenHousePressure, double potPotassium, double potPhosphorus, double potNitrogen) {
        this.potId = potId;
        this.potName = potName;
        this.potType = potType;
        this.plantName = plantName;
        this.userId = userId;
        this.hasCamera = hasCamera;
        this.pictReq = pictReq;
        this.pumpStatus = pumpStatus;
        this.greenHouseStatus = greenHouseStatus;
        this.greenHouseTemperature = greenHouseTemperature;
        this.greenHouseHumidity = greenHouseHumidity;
        this.greenHousePressure = greenHousePressure;
        this.potPotassium = potPotassium;
        this.potPhosphorus = potPhosphorus;
        this.potNitrogen = potNitrogen;
    }

    // Getters
    public int getPotId() {
        return potId;
    }

    public String getPotName() {
        return potName;
    }

    public int getPotType() {
        return potType;
    }

    public String getPlantName() {
        return plantName;
    }

    public int getUserId() {
        return userId;
    }

    public boolean hasCamera() {
        return hasCamera;
    }

    public boolean isPictReq() {
        return pictReq;
    }

    public boolean isPumpStatus() {
        return pumpStatus;
    }

    public boolean isGreenHouseStatus() {
        return greenHouseStatus;
    }

    public double getGreenHouseTemperature() {
        return greenHouseTemperature;
    }

    public double getGreenHouseHumidity() {
        return greenHouseHumidity;
    }

    public double getGreenHousePressure() {
        return greenHousePressure;
    }

    public double getPotPotassium() {
        return potPotassium;
    }

    public double getPotPhosphorus() {
        return potPhosphorus;
    }

    public double getPotNitrogen() {
        return potNitrogen;
    }

    public String getPlant() {
        return plantName;
    }
}

File: SQLserverFinale/SQLserverFinale/Models/Pot.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.Data.SqlClient;
using Microsoft.EntityFrameworkCore;
using System.Diagnostics.CodeAnalysis;

namespace SQLserverFinale.Models
{
    [Index(nameof(Pot.PotName), IsUnique = true)] 
    public class Pot
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int PotId { get; set; }
        [Required,NotNull]
        public string PotName { get; set; }
        [Required]
        public int PotType { get; set; }
        [ForeignKey("Plant")]
        [MaxLength(100)]
        public string PlantName { get; set; }

        internal static object FromSqlRaw(string v, SqlParameter sqlParameter)
        {
            throw new NotImplementedException();
        }
        [ForeignKey("User")]
        [Required]
        public int UserId { get; set; }
        public bool? HasCamera { get; set; }
        public bool? PictReq { get; set; }
        public bool? PumpStatus { get; set; }
        public bool? GreenHouseStatus { get; set; }
        public double? GreenHouseTemperature { get; set; }
        public double? GreenHouseHumidity { get; set; }
        public double? GreenHousePressure { get; set; }
        public double? PotPotassium {  get; set; }
        public double? PotPhospor {  get; set; }
        public double? PotNitrogen { get; set; }
        // public Plant Plant { get; set; }

    }
}
